### PR TITLE
Improve unit test coverages for Devices V2

### PR DESCRIPTION
**Pull request by schoims**

### CODE REVIEW

File: iothub/service/tests/DeviceAuthenticationTests.cs [hunk 1]
```diff
@@ -241,13 +241,13 @@ public async Task DeviceAuthenticationGoodAuthConfigTest6()
                 mockHttpRequestFactory,
                 s_retryHandler);
 
-            await devicesClient.CreateAsync(deviceBadAuthConfig).ConfigureAwait(false);
+            await devicesClient.CreateAsync(deviceGoodAuthConfig).ConfigureAwait(false);
         }
 
         [TestMethod]
-        public async Task DeviceAuthenticationGoodAuthSHA256()
+        public async Task DeviceAuthentication_CertificateAuthConfigTest4_SHA256()
```
Comment (drwill-ms) [inline iothub/service/tests/DeviceAuthenticationTests.cs:248]:
```suggestion
        public async Task DeviceAuthentication_CertificateAuthConfigTest4_Sha256()
```

File: iothub/service/tests/DeviceAuthenticationTests.cs [hunk 2]
```diff
@@ -278,13 +278,13 @@ public async Task DeviceAuthenticationGoodAuthSHA256()
                 mockHttpRequestFactory,
                 s_retryHandler);
 
-            await devicesClient.CreateAsync(deviceBadAuthConfig).ConfigureAwait(false);
+            await devicesClient.CreateAsync(deviceGoodAuthConfig).ConfigureAwait(false);
         }
 
         [TestMethod]
-        public async Task DeviceAuthenticationIsCertificateAuthority()
+        public async Task DeviceAuthentication_CertificateAuthConfigTest5()
```
Comment (drwill-ms) [inline iothub/service/tests/DeviceAuthenticationTests.cs:285]: Can we come up with a meaningful name about the case this test covers without having to use a numeric suffix?

File: iothub/service/tests/Registry/DevicesClientTests.cs [hunk 3]
```diff
@@ -177,6 +180,134 @@ public async Task DevicesClient_CreateAsync_EmptyListThrows()
...
+        }
+
+        [TestMethod]
+        public async Task DevicesClient_CreateWithTwinAsync()
+        {
+            // arrange
+            var TestTagName = "DevicesClient_Tag";
+            var TestTagValue = 100;
+            var goodDevice1 = new Device("123") { ConnectionState = ClientConnectionState.Connected };
+            ClientTwin clientTwin1 = new ClientTwin("123")
```
Comment (drwill-ms) [inline iothub/service/tests/Registry/DevicesClientTests.cs:225]:
```suggestion
            var clientTwin1 = new ClientTwin("123")
```

File: iothub/service/tests/Registry/DevicesClientTests.cs [hunk 4]
```diff
@@ -177,6 +180,134 @@ public async Task DevicesClient_CreateAsync_EmptyListThrows()
...
+
+            // assert
+            await act.Should().NotThrowAsync().ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task DevicesClient_CreateWithTwinAsync()
+        {
+            // arrange
+            var TestTagName = "DevicesClient_Tag";
```
Comment (drwill-ms) [inline iothub/service/tests/Registry/DevicesClientTests.cs:222]: local constants should use camelCase

File: iothub/service/tests/Registry/DevicesClientTests.cs [hunk 5]
```diff
@@ -558,5 +689,488 @@ public async Task DevicesClient_DeleteAsync_OnlyIfUnchangedTrueHasEtags()
...
+            // arrange
+            var jobId = "sampleJob";
+            var jobStatus = JobStatus.Completed;
+
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            mockCredentialProvider
+                .Setup(getCredential => getCredential.GetAuthorizationHeader())
+                .Returns(s_validMockAuthenticationHeaderValue);
+            var mockHttpRequestFactory = new HttpRequestMessageFactory(s_httpUri, "");
+            var jobsToReturn = new List<IotHubJobResponse>() { new IotHubJobResponse() { JobId = jobId, Status = jobStatus } };
```
Comment (drwill-ms) [inline iothub/service/tests/Registry/DevicesClientTests.cs:810]: You can drop the ctor parens when using an initializer list.

```suggestion
            var jobsToReturn = new List<IotHubJobResponse> { new IotHubJobResponse { JobId = jobId, Status = jobStatus } };
```

File: iothub/service/tests/SerializationTests.cs [hunk 6]
```diff
@@ -24,7 +25,8 @@ public void Twin_JsonDateParse_Ok()
  ""lastActivityTime"": ""2018-06-29T21:17:08.7759733"",
 }";
 
-            JsonConvert.DeserializeObject<ClientTwin>(jsonString);
+            ClientTwin clientTwin = JsonConvert.DeserializeObject<ClientTwin>(jsonString);
+            clientTwin.DeviceId.Should().Be("test");
```
Comment (drwill-ms) [inline iothub/service/tests/SerializationTests.cs:29]: If you can, make a constant for this so we don't have to rely on static strings matching.

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/tests/DeviceAuthenticationTests.cs (hunk 2 above)
Comment (schoims) [reply iothub/service/tests/DeviceAuthenticationTests.cs:285]: Absolutely!

**Pull request merged**